### PR TITLE
Update documentation for new API.

### PR DESCRIPTION
**Pull request by chris-horner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,83 +14,74 @@ We have found this particularly useful on Android, where there are [many options
...
 ```java
 public final class Person {
-	public final String name;
-	public final int age;
+  public final String name;
+  public final int age;
 }
 ```
 
 To persist a single `Person`, we must first create `ValueStore`.
```
Comment (stphven) [inline README.md]: Missing an 'a' - should be ``we must first create a `ValueStore`.``

File: rxstore/src/main/java/au/com/gridstone/rxstore/ValueStore.java [hunk 2]
```diff
@@ -23,26 +23,75 @@
...
+   * immediately with the current value upon subscription and subsequent changes thereafter.
+   * <p>
+   * Values are wrapped inside of {@link ValueUpdate ValueUpdate} objects, as it's possible for a
+   * store to hold no current value. If that's the case {@link ValueUpdate#empty update.empty} will
+   * be true and {@link ValueUpdate#value update.value} will be null.
+   */
   @NonNull Observable<ValueUpdate<T>> observe();
 
+  /**
+   * Clear the value in this store and observe the operation. (Useful for chaining).
```
Comment (jsprincep) [inline rxstore/src/main/java/au/com/gridstone/rxstore/ValueStore.java]: nit: I personally prefer this `(Useful for chaining)` style over the earlier `making this useful for chaining` comments, which felt a bit a clunky.

Comment (chris-horner) [inline rxstore/src/main/java/au/com/gridstone/rxstore/ValueStore.java]: Yeah, it was a tricky call. I felt the aside in parentheses wasn't quite clear enough and wanted it to have some more emphasis. Happy to change it back though.

Comment (jsprincep) [inline rxstore/src/main/java/au/com/gridstone/rxstore/ValueStore.java]: Personally I prefer the comment in parentheses, but it is totally subjective, and up to you.

**Pull request merged**